bslib.brs

```diff
@@ -21,4 +21,27 @@ function bslib_toString(value)
 
     'when all else fails, just return the type
     return "<" + valueType + ">"
+end function
+
+'
+' Simple ternary function. Given a condition, return the
+' consequent if true, and the alternate if false
+'
+function bslib_ternary(condition, consequent, alternate)
+    if condition then
+        return consequent
+    else
+        return alternate
+    end if
+end function
+
+'
+' Return consequent if consequent is not invalid, otherwise return alternate
+'
+function bslib_coalesce(consequent, alternate)
+    if consequent <> invalid then
```

Comment:
Can the `consequent` be "uninitialized" as opposed to `invalid`?

Reply:
I spoke with @georgejecook and @triwav about this a bit, and we decided we agreed that none of us think the `null coalescing operator` should support `<uninitialized>`. Any object property access for undefined properties will return `invalid`, so there are very few cases where `<uninitialized>` could actually show up:
1. typing the name of a variable that was not defined
2. using the name of a loop variable before its first assignment.


Both of these are edge cases should happen very infrequently in the wild. In almost all cases, getting `uninitialized` is a program error, which is technically different than `invalid`.  

Here are the reasons we don't want to support `<uninitialized>`:
1. it will add a `type()` call to every ternary operator, which will incur a performance cost.
2. `<uninitialized>` is a program error, and _should_ cause a runtime exception in my opinion
3. it's such an edge case, it shouldn't happen all that frequently
4. type checking (or `bslint`) should catch both the use-before-initialized or unknown-variable during the compile phase

Thoughts?

Reply:
Sounds good.